package/MDAnalysis/coordinates/CHEMFILES.py

```diff
@@ -0,0 +1,201 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+TODO
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import numpy as np
+
+from . import base, core
+
+from chemfiles import Trajectory, Frame, Atom, UnitCell, Residue
+
+class ChemfilesReader(base.ReaderBase):
+    """
+    TODO
+    """
+    format = 'CHEMFILES'
+    units = {'time': 'fs', 'length': 'Angstrom'}
+
+    def __init__(self, filename, format="", **kwargs):
+        """
+        Parameters
+        ----------
+        filename : str
+            trajectory filename
+        format : str (optional)
+            use the given format name instead of guessing from the extension
+            TODO: link to the doc
+        **kwargs : dict
+            General reader arguments.
+        """
+        super(ChemfilesReader, self).__init__(filename, **kwargs)
+        self._format = format
+        self._cached_n_atoms = None
+        self._open()
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+        self.next()
+
+    def _open(self):
+        self._file = Trajectory(self.filename, 'r', self._format)
+        self._closed = False
+        self._step = 0
```

Comment:
Adding `self._frame = -1` here allow the failing tests about auxiliaries to pass. I still need to figure out why, though.

It does not fix the test about containers.